src/scheduler/pbs_sched.c

```diff
@@ -159,6 +160,11 @@ extern char *msg_startup1;
 void
 on_segv(int sig)
 {
+	int *thid;
+
+	thid = (int *) pthread_getspecific(th_id_key);
+	if (thid != NULL && *thid != 0)
+		pthread_exit(NULL);
```

Comment:
Does this mean that a specific thread will exit? If so, then upon receiving SEGV more than 5 min apart we fork() ourselves and let the child continue. 
How will threads behave when scheduler forks and wouldn't pthread_exit() reduce a worker thread out of the process and scheduler will not be aware of this.

Reply:
this will make all worker threads exit, it's a bad idea to fork() with threads, it's safer to create them again, which should happen when the new process executes afresh. But I noticed that I'm not calling kill_threads() here, which should get called before the fork() as well.